src/components/Budget/BudgetForm/BudgetExpenseClassesFields/BudgetExpenseClassesFields.js

```diff
@@ -0,0 +1,116 @@
+import React, { useMemo, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { FieldArray } from 'react-final-form-arrays';
+
+import {
+  Col,
+  Label,
+  RepeatableField,
+  Row,
+  TextField,
+} from '@folio/stripes/components';
+import {
+  FieldSelectFinal,
+  FieldSelectionFinal,
+  validateRequired,
+} from '@folio/stripes-acq-components';
+
+import {
+  EXPENSE_CLASS_STATUS,
+  EXPENSE_CLASS_STATUS_OPTIONS,
+} from '../../constants';
+
+const headLabels = (
+  <Row>
+    <Col xs>
+      <Label
+        id="budgetExpenseClassesFormNameLabel"
+        required
+      >
+        <FormattedMessage id="ui-finance.budget.expenseClasses.expenseClassName" />
+      </Label>
+    </Col>
+
+    <Col xs>
+      <Label id="budgetExpenseClassesFormAccountNumberLabel">
+        <FormattedMessage id="ui-finance.budget.expenseClasses.accountNumber" />
+      </Label>
+    </Col>
+
+    <Col xs>
+      <Label id="budgetExpenseClassesFormStatusLabel">
+        <FormattedMessage id="ui-finance.budget.expenseClasses.status" />
+      </Label>
+    </Col>
+  </Row>
+);
+
+const BudgetExpenseClassesFields = ({ expenseClasses, formValues }) => {
+  const expenseClassesOptions = useMemo(
+    () => expenseClasses.map(({ id, name }) => ({
+      label: name,
+      value: id,
+    })),
+    [expenseClasses],
+  );
+
+  const addFields = useCallback(fields => fields.push({ status: EXPENSE_CLASS_STATUS.ACTIVE }), []);
+
+  const renderSubForm = useCallback((field, index) => {
+    const getAccountNumberExt = () => (
+      expenseClasses.find(({ id }) => (
+        id === formValues.statusExpenseClasses[index].expenseClassId
+      ))?.externalAccountNumberExt ?? null
+    );
+
+    return (
+      <Row>
+        <Col xs>
+          <FieldSelectionFinal
+            ariaLabelledBy="budgetExpenseClassesFormNameLabel"
+            dataOptions={expenseClassesOptions}
+            fullWidth
+            name={`${field}.expenseClassId`}
+            required
+            validate={validateRequired}
+          />
+        </Col>
+        <Col xs>
+          <TextField
+            ariaLabelledBy="budgetExpenseClassesFormAccountNumberLabel"
+            value={getAccountNumberExt()}
+            disabled
+          />
+        </Col>
+        <Col xs>
+          <FieldSelectFinal
+            aria-labelledby="budgetExpenseClassesFormStatusLabel"
+            dataOptions={EXPENSE_CLASS_STATUS_OPTIONS}
+            fullWidth
+            name={`${field}.status`}
+          />
+        </Col>
+      </Row>
+    );
+  }, [expenseClasses, formValues]);
```

Comment:
you shouldn't depend on `formValues` actually. Just extract what you need above and use only that thing

Reply:
fixed